cms/djangoapps/contentstore/views/tests/test_videos.py

```diff
@@ -499,6 +502,39 @@ def test_assume_role_connection(self, mock_conn, mock_key):
                 security_token=credentials['session_token']
             )
 
+    @patch('boto.s3.key.Key')
+    @patch('boto.s3.connection.S3Connection')
+    @override_flag(waffle_flags()[ENABLE_VEM_PIPELINE].namespaced_flag_name, active=True)
+    def test_enable_vem_pipeline(self, mock_conn, mock_key):
+        """
+        Test that if VEM pipeline is enabled, objects are uploaded to the correct s3 bucket
+        """
+        files = [{'file_name': 'first.mp4', 'content_type': 'video/mp4'}]
+
+        bucket = Mock()
+        mock_conn.return_value = Mock()
+        mock_conn.return_value.get_bucket = Mock(return_value=bucket)
+        mock_key_instances = [
+            Mock(
+                generate_url=Mock(
+                    return_value='http://example.com/url_{}'.format(file_info['file_name'])
+                )
+            )
+            for file_info in files
+        ]
+        mock_key.side_effect = mock_key_instances + [Mock()]
```

Comment:
Why the extra mock append?

Reply:
I saw it in other test cases and from what I could gather, it is added as a fail-safe if extra calls are made.